src/transpile/fusion.hpp

```diff
@@ -312,204 +370,41 @@ bool Fusion::aggregate_operations(oplist_t& ops, const int fusion_start, const i
 
     if (to != i) {
       std::vector<op_t> fusioned_ops;
+      std::set<uint_t> fusioned_qubits;
       for (int j = to; j <= i; ++j) {
         fusioned_ops.push_back(ops[j]);
+        std::copy(ops[j].qubits.cbegin(), ops[j].qubits.cend(),
+                  std::inserter(fusioned_qubits, fusioned_qubits.end()));
```

Comment:
```suggestion
        fusioned_qubits.insert(ops[j].qubits.cbegin(), ops[j].qubits.cend())
```
More compact and probably faster